src/twister2/yaml_test_function.py

```diff
@@ -82,5 +82,8 @@ def __call__(
 
                 assert test.result == SubTestStatus.PASS, f'Subtest {test.testname} failed'
 
-        failed_msg: str = 'Test failed due to: {}'.format('\n'.join(log_parser.messages))
+        if log_parser.state == log_parser.STATE.UNKNOWN:
+            failed_msg: str = f'Test state is {log_parser.state.value}'
```

Comment:
To increase readability of this message perhaps there should be this additional info about timeout:
```python
failed_msg: str = f'Test state is {log_parser.state.value} (timeout has probably occurred)'
```

Reply:
Changed